chain-abci/src/enclave_bridge/real/mod.rs

```diff
@@ -0,0 +1,91 @@
+mod enclave_u;
+mod server;
+#[cfg(feature = "sgx-test")]
+pub mod test;
+
+use crate::enclave_bridge::real::enclave_u::{check_initchain, check_tx, end_block};
+use crate::enclave_bridge::EnclaveProxy;
+use chain_storage::ReadOnlyStorage;
+use enclave_protocol::{IntraEnclaveRequest, IntraEnclaveResponse};
+use enclave_u_common::enclave_u::init_enclave;
+use log::info;
+use server::TxValidationServer;
+use sgx_urts::SgxEnclave;
+use std::sync::mpsc::channel;
+use std::thread;
+
+pub const TX_VALIDATION_ENCLAVE_FILE: &str = "tx_validation_enclave.signed.so";
+
+pub struct TxValidationApp {
+    enclave: SgxEnclave,
+}
+
+impl TxValidationApp {
+    pub fn new() -> Self {
+        info!("Attempting to launch TX Validation Enclave in debug mode");
+        let enclave = match init_enclave(TX_VALIDATION_ENCLAVE_FILE, true) {
+            Ok(r) => {
+                info!(
+                    "[+] Init TX Validation Server Enclave Successful {}!",
+                    r.geteid()
+                );
+                r
+            }
+            Err(x) => {
+                panic!(
+                    "[-] Init  TX Validation  Sercer Enclave Failed {}!",
+                    x.as_str()
+                );
+            }
+        };
+        Self { enclave }
+    }
+}
+
+impl EnclaveProxy for TxValidationApp {
+    fn check_chain(&self, network_id: u8) -> Result<(), ()> {
+        check_initchain(self.enclave.geteid(), network_id)
+    }
+
+    fn process_request(&mut self, request: IntraEnclaveRequest) -> IntraEnclaveResponse {
+        let eid = self.enclave.geteid();
+        match &request {
+            IntraEnclaveRequest::EndBlock => end_block(eid, request),
+            IntraEnclaveRequest::Encrypt(_) => {
+                unreachable!("should be used only in TxValidationServer")
+            }
+            IntraEnclaveRequest::ValidateTx { .. } => check_tx(eid, request),
+        }
+    }
+}
+
+pub fn start_zmq(
+    zmq_conn_str: &str,
+    network_id: u8,
+    storage: ReadOnlyStorage,
+) -> thread::JoinHandle<()> {
+    info!("Attempting to launch TX Validation Enclave (for tx-query / zmq) in debug mode");
+    let enclave = match init_enclave(TX_VALIDATION_ENCLAVE_FILE, true) {
```

Comment:
adding some comment whether this is thread or process ?

Reply:
ok